src/content/content.npmjs.js

```diff
@@ -8,15 +8,33 @@ const addPackageReport = (packageID) => {
   packageReport.setAttribute('package-name', packageID.name);
 
   const repository = document.querySelector('#repository');
-  if (repository) {
-    repository.parentElement.insertBefore(packageReport, repository);
+
+  const config = { attributes: false, childList: true, subtree: true };
+  const observer = new MutationObserver(() => {
+    if (document.contains(repository)) {
+      repository.parentElement.insertBefore(packageReport, repository);
+      observer.disconnect();
+    }
+  });
+
+  observer.observe(document, config);
```

Comment:
Why?

Reply:
There are cases where the `repository` element is being rendered after the querySelector queries the document (for example - searching for package `x` in npmjs' home site and being redirected to `/package/x`). 
In those cases, using the "regular" querySelector will return `undefined` and won't render the tooltip.

Reply:
It is hard for me to test it right now, So I will trust you with this (I faced a similar problem in ChatGPT), and I'm asking you

1. Are you sure you need to observe the `document`? Maybe you can observe more specific element, for example, `main div.w-third-l`?
2. I can't understand if the bug is when `repository = document.querySelector('#repository')` does not exist, with your code it will still be `null`, and the condition `document.contains(repository)` will be always `false`.
  In other words, I think you need to move the `repository = document.querySelector('#repository')` into the `MutationObserver`.
3. Can you please 🙏🙏🙏 write an E2E test for this case too? It will protect us from breaking such cases of missing events or URL changes.
  I'm asking it because, as I said, I sew problems like that before, and I didn't have the confidence to handle them, because I was not covered by tests.